agent/discovery/sapsystem.go

```diff
@@ -33,33 +33,49 @@ func (d SAPSystemsDiscovery) Discover() error {
 		return err
 	}
 
+	sapSystems := make([]*sapsystem.SAPSystem, len(systems))
 	d.SAPSystems = systems
-	for _, s := range d.SAPSystems {
+	for i, s := range d.SAPSystems {
 		err := s.Store(d.discovery.client)
 		if err != nil {
 			return err
 		}
 
-		// Store SAP System, Landscape and Environment names on hosts metadata
-		err = storeSAPSystemTags(d.discovery.client, s)
-		if err != nil {
-			return err
-		}
+		sapSystems[i] = s
+	}
+
+	// Store SAP System, Landscape and Environment names on hosts metadata
+	err = storeSAPSystemTags(d.discovery.client, sapSystems)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func storeSAPSystemTags(client consul.Client, system *sapsystem.SAPSystem) error {
-	envName, landName, sysName, err := loadSAPSystemTags(client, system.SID)
-	if err != nil {
-		return err
+func storeSAPSystemTags(client consul.Client, systems []*sapsystem.SAPSystem) error {
+	var envName, landName string
+	sysNames := ""
+	var err error
+	for i, system := range systems {
```

Comment:
This looks a bit over complicated.
I suggest the next:
As `envName` and `landName` will be the same independtly which `system.SID` we use, we could just call it once with the 1st SID.
After that, i would prefer the usage of `strings.Join` to create the sids string. I don't know if we could use some kind of lamdba method to get the SID, otherwise we will need to loop though all the systems, create a SID string array, and apply the `Join` function